### ISSUE TITLE
Backslash in (access) key leads to problems with URI

### ISSUE THREAD

**Issue by golsch** [carlspring/s3fs-nio#771]
# Bug Description

Unfortunately, our S3 implementation is dependent on another system, which means that there is a backslash in the access key.
If you create an absolute URI for a path, this leads to problems because backslash is an illegal URI character.

# Steps To Reproduce

1. Initialize an S3 file system with a backslash in the access key, for example <ad\accesskey>
2. Create an absolute path or URI for a path

# Expected Behavior

Get an absolute path/URI

# Question

Is the user info (<>@) in the URI even necessary?

**Comment by steve-todorov** [comment]
Hey @golsch ,

Doesn't escaping the `\` access key character with `\\` work for this case?

Re:question:

TL;DR - yes, it's necessary for the time being. You might want to have multiple instances that use different buckets, so yes. We have plans to refactor this at some point so it's a bit more convenient to use, but for now it's necessary.
